### ISSUE TITLE
Separate XServ into standalone binary

### ISSUE THREAD

**Issue by filfat** [codyd51/axle#26]
Eventually preferably XServ should be it's own standalone binary. This would of course require a way to load and run external binaries but also a way to access some kind of an axle api (for gfx etc, would also require #18 ).

**Comment by codyd51** [comment]
I completely agree. axle takes 'monolithic kernel' to the logical extreme, and I'd like to finally get ELF loading working so we can refactor xserv (along with almost every other shell program) into their own binaries.

I have some notion of how I'll want to implement IPC once we get to this point.

**Comment by codyd51** [comment]
Just an update on this, I've ported newlib to axle; it includes stubs for sys calls axle doesn't support and functionality for ones it does. Also, I've implemented an ELF loader that tends to work... most of the time... if you're lucky. Needless to say, both components need work. However, external kernel modules/userspace programs are getting closer and closer to reality :)
